src/sagemaker/image_uris.py

```diff
@@ -248,6 +269,20 @@ def config_for_framework(framework):
         return json.load(f)
 
 
+def _get_inference_tool(inference_tool, instance_type):
+    """Extract the inference tool name from instance type."""
+    if not inference_tool and instance_type:
+        match = re.match(r"^ml[\._]([a-z\d]+)\.?\w*$", instance_type)
```

Comment:
what happens if there is one match here

Reply:
if there is not match, then `inference_tool` is returned (the 1st input argument of this function).

Reply:
you are checking match[1], will you always have match[1]

Reply:
yes. match[0] is the full input string. in this case `instance_type`

Reply:
```
>>> import re
>>> instance_type = "ml.inf.xlarge"
>>> match = re.match(r"^ml[\._]([a-z\d]+)\.?\w*$", instance_type)
>>> match[0]
'ml.inf.xlarge'
>>> match[1]
'inf'

```